### ISSUE TITLE
Calculation doesn't work when the .ffn file contains only 1 sequence

### ISSUE THREAD

**Issue by kyu999** [mariodosreis/tai#1]
Hello, 

I've read [your paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC521650/) and very impressed, so I am trying to use tAI value for codon optimization in order to increase protein expression.
The instruction of README guided me well, made me calculate tAI value easily.
However, while calculation I got some error, that may be something I should not ignore.
The procedure is as follows:

`$ perl  codonM <MyGeneToExpress>.fasta <MyGeneToExpress>.m`
`$ R`
`> require("tAI")`
`> eco.trna <- scan("ecolik12.trna")`
`> eco.ws <- get.ws(tRNA=eco.trna, sking=1)`
`> eco.m <- matrix(scan("<MyGeneToExpress>.m"), ncol=61, byrow=TRUE)`
`> eco.tai <- get.tai(eco.m, eco.ws)`
`Error at apply(x, 1, "*", w) : dim(X) must have a positive length`

I revealed that the error happened at this line and such calculation looks occur.
https://github.com/mariodosreis/tai/blob/master/R/tAI.R#L193

At this point, I doubted that the variable `w` and `x`(`eco.m` in that case) are inconsistent for calculate.
Therefore I look closely how the content of each variable were changing.
Curiously, the shape of `eco.m` drastically changed after removing methionine.

`> eco.m <- matrix(scan("<MyGeneToExpress>.m"), ncol=61, byrow=TRUE)`
`Read 61 items`
`> eco.m`
     [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13] [,14]
[1,]   21   13    0    0    9   11   13    8   17    25     0     0     0     0
     [,15] [,16] [,17] [,18] [,19] [,20] [,21] [,22] [,23] [,24] [,25] [,26]
[1,]     0     0     0    23    13    35    14     4     2     0     0     0
     [,27] [,28] [,29] [,30] [,31] [,32] [,33] [,34] [,35] [,36] [,37] [,38]
[1,]     0     0     0     8    10    17     0     0     0     0     0     1
     [,39] [,40] [,41] [,42] [,43] [,44] [,45] [,46] [,47] [,48] [,49] [,50]
[1,]     2     0     0    10     7     0     0     0     9    20     5    39
     [,51] [,52] [,53] [,54] [,55] [,56] [,57] [,58] [,59] [,60] [,61]
[1,]    28    27    23     0     0     0     0    44    51    36    55
`> eco.m <- eco.m[,-33]`
`> eco.m`
 [1] 21 13  0  0  9 11 13  8 17 25  0  0  0  0  0  0  0 23 13 35 14  4  2  0  0
[26]  0  0  0  0  8 10 17  0  0  0  0  1  2  0  0 10  7  0  0  0  9 20  5 39 28
[51] 27 23  0  0  0  0 44 51 36 55

In short, `eco.m` changed its shape from matrix to vector I guess, so
I added a bit processing to avoid this error.

`> eco.m <- eco.m[,-33]`
`> eco.m <- t(matrix(eco.m))` # added
`> get.tai(eco.m, eco.ws)`
`[1] 0.2511931`

So my question is, do you think this processing is ok to add?

**Comment by mariodosreis** [comment]
I see. R automatically demotes a matrix to a vector if the matrix only has one row. That was the problem. I didn’t run your code but your correction seems sensible.

Best,

Mario dos Reis

Lecturer @ QMUL
dosreis.github.io

**Comment by mariodosreis** [comment]
Try this:

> require("tAI")
> eco.trna <- scan("ecolik12.trna")
> eco.ws <- get.ws(tRNA=eco.trna, sking=1)
> eco.m <- matrix(scan("<MyGeneToExpress>.m"), ncol=61, byrow=TRUE)
> eco.m <- eco.m[,-33, drop=F]. # This is the difference!!!!!
> eco.tai <- get.tai(eco.m, eco.ws)

And let me know if it works.

Best,

Mario dos Reis

Lecturer @ QMUL
dosreis.github.io

**Comment by kyu999** [comment]
Thank you for your response!
It worked well :)
